src/windowsservercore/ltsc2022/helix/webassembly-net8/amd64/Dockerfile

```diff
@@ -15,36 +17,41 @@ COPY arial.ttf c:/windows/fonts
 ENV GIT_VERSION=2.34.1
 ENV GIT_INSTALLER=MinGit-${GIT_VERSION}-64-bit.zip
 
-RUN curl -SL --output %TEMP%\%GIT_INSTALLER% https://github.com/git-for-windows/git/releases/download/v%GIT_VERSION%.windows.1/%GIT_INSTALLER% \
-    && mkdir C:\git \
-    && tar -C C:\git -zxf %TEMP%\%GIT_INSTALLER% \
+RUN curl -SL --output %TEMP%\%GIT_INSTALLER% https://github.com/git-for-windows/git/releases/download/v%GIT_VERSION%.windows.1/%GIT_INSTALLER% `
+    && mkdir C:\git `
+    && tar -C C:\git -zxf %TEMP%\%GIT_INSTALLER% `
     && setx PATH "%PATH%;C:\git\cmd"
 
 # fix certificates for python to be able to download emscripten files
 RUN certutil -generateSSTFromWU roots.sst && certutil -addstore -f root roots.sst && del roots.sst
 
 # Install Emscripten toolchain
-ENV EMSCRIPTEN_VERSION=3.1.34
-ENV EMSCRIPTEN_PATH="C:\emscripten"
-ENV EMSDK_PATH="C:\emscripten\emsdk"
-
-RUN mkdir %EMSCRIPTEN_PATH% \
-    && cd %EMSCRIPTEN_PATH% \
-    && git clone https://github.com/emscripten-core/emsdk.git %EMSDK_PATH%
-RUN cd %EMSDK_PATH% \
-    && git checkout %EMSCRIPTEN_VERSION% \
-    && .\emsdk install %EMSCRIPTEN_VERSION%-upstream  \
-    && .\emsdk activate %EMSCRIPTEN_VERSION%-upstream \
+ENV `
+    EMSCRIPTEN_VERSION=3.1.34 `
+    EMSCRIPTEN_PATH="C:\emscripten" `
+    EMSDK_PATH="C:\emscripten\emsdk"
+
+RUN mkdir %EMSCRIPTEN_PATH% `
+    && cd %EMSCRIPTEN_PATH% `
+    && git clone https://github.com/emscripten-core/emsdk.git %EMSDK_PATH% `
+    && cd %EMSDK_PATH% `
+    && git checkout %EMSCRIPTEN_VERSION% `
+    && .\emsdk install %EMSCRIPTEN_VERSION%-upstream  `
+    && .\emsdk activate %EMSCRIPTEN_VERSION%-upstream `
     && .\upstream\emscripten\embuilder.bat build MINIMAL
 
-# install Node JS
-ENV NODE_VERSION 18.20.5
-
-RUN curl -SL --output %TEMP%\nodejs.msi https://nodejs.org/dist/v%NODE_VERSION%/node-v%NODE_VERSION%-x64.msi
-RUN msiexec /i %TEMP%\nodejs.msi /quiet /passive /qn /norestart
+# Install latest 18.x Node JS
+RUN powershell -Command " `
+        $ErrorActionPreference = 'Stop'; `
+        $ProgressPreference = 'SilentlyContinue'; `
+        `
+        $json = Invoke-RestMethod -Uri 'https://nodejs.org/dist/index.json'; `
+        $nodeVersion = ($json | ForEach-Object { if ($_.version -match '^v18\.') { $_ } } | Sort-Object -Property version -Descending | Select-Object -First 1).version.TrimStart('v'); `
```

Comment:
Using a regex here seems unnecessary. Using `$_.version.StartsWith("v18")` seems a bit more clear.

Reply:
Fixed in https://github.com/dotnet/dotnet-buildtools-prereqs-docker/pull/1257/commits/95c5962c76c4420b9f674fec6f9c6ab7d7ae61a9